core/app/models/spree/stock/estimator.rb

```diff
@@ -19,14 +19,26 @@ def initialize(order)
       def shipping_rates(package, frontend_only = true)
         rates = calculate_shipping_rates(package)
         rates.select! { |rate| rate.shipping_method.frontend? } if frontend_only
-        choose_default_shipping_rate(rates)
+        choose_selected_shipping_rate(rates)
         sort_shipping_rates(rates)
       end
 
       private
-      def choose_default_shipping_rate(shipping_rates)
-        unless shipping_rates.empty?
-          shipping_rates.min_by(&:cost).selected = true
+      def choose_selected_shipping_rate(shipping_rates)
+        # unless shipping_rates.empty?
+        #   shipping_rates.min_by(&:cost).selected = true
+        # end
+        return if shipping_rates.empty?
+        shipping_rate = find_preselected_shipping_rate(shipping_rates) || shipping_rates.min_by(&:cost)
+        shipping_rate.selected = true
+      end
+
+      def find_preselected_shipping_rate(shipping_rates)
+        if order.shipments.count > 0
+          shipping_method = order.shipments.map {|s| s.selected_shipping_rate.try!(:shipping_method) }.compact.uniq.first
```

Comment:
Yes. `refresh_rates` is slightly odd in that it only uses the `Stock::Estimator` and not the `Stock::Coordinator` but since this logic is in the Estimator, we're good.